include/knowhere/dataset.h

```diff
@@ -162,6 +168,17 @@ class DataSet : public std::enable_shared_from_this<const DataSet> {
         return nullptr;
     }
 
+    const void*
+    GetCentroidIdMapping() const {
```

Comment:
And these Getter. We can have a index_dataset_util & another cluster_dataset_util to gather them.